README.md

```diff
@@ -1 +1,291 @@
+2. Canvas B, which does not support Feature X, opens the Open Canvas file, and some edits are made to the canvas.
+3. Canvas B exports the canvas to an Open Canvas file. The nodes for Feature X should still be in the Open Canvas file, unchanged.
+
+NOTE: This is perhaps the most onerous part of supporting Open Canvas; implementing canvases have to store data that they don't need or recognize.
+
+### Extensibility through fallbacks
+
+The base node type supports an optional `fallback` property as a string. If a canvas wishes to visualize a node that it doesn't implement, the canvas can render a fallback element on the canvas at the element's x,y position using the string.
+
+### Extensibility through schema extension
+
+Schemas can extend one another.
+
+When extending, properties are simply merged to establish the full schema. This allows extension with meaningful fallbacks: if an implementer doesn't support the extended schema, but does support the base schema, they can render the node using a simpler representation.
+
+Schemas are defined in JSONSchema format.
+
+```json
+"schemas": [
+  "@ocwg/text": { /* JSONSchema format */ }
+]
+```
+
+For example, a `@tldraw/arrow` schema could extend a `@ocwg/arrow` schema, adding properties that are only applicable to TLDraw arrow.
+
+- [ ] Flesh out this example
+
+## Core Node Schemas
+
+These are the node schemas included in the core `@ocwg` namespace.
+
+### Text
+
+Text Nodes are used to display text on the canvas.
+
+### Assets
+
+Asset Nodes are used to represent files or other assets. They can be embedded or reference external sources. We use `mime-type` to describe the type of the asset and `uri` to describe the location of the asset.
+
+### Arrows
+
+### Examples
+
+Here are a bunch of example Nodes:
+
+```json
+{
+  "nodes": [
+    {
+      "id": "someShapeID11321",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/text",
+        "schema_version": "1.0",
+        "text": "Hello, world!",
+        "color": "#000000"
+      }]
+    },
+    {
+      "id": "2435234634t324t3245",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/text",
+        "schema_version": "1.0",
+        "w": 200,
+        "h": 150,
+        "mime_type": "text/json",
+        "uri": "data://9823fy9283hf2i3yf082ohu3fo2hufi2uh4fk234f8youhi",
+      }]
+    },
+    {
+      "id": "oajefioajef83r8y3ehf",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/rectangle",
+        "schema_version": "1.0",
+        "width": 200,
+        "height": 150,
+        "text": "Hello, world!",
+        "stroke_color": "red",
+        "stroke_weight": ,
+        "fill_color": ,
+        "rotation": 90 /* +/-360 degrees */
+      }]
+    },
+    {
+      "id": "someShapeID124135",
+      "x": 10,
+      "y": 10,
+      "properties": [{
```

Comment:
It wouldn't be a stretch to assume that "x" and "y" are position numbers, but this could be made more explicit by calling the property "position". You could also have an X scale or a Y scale, for example.

Also, it's interesting to see "x" and "y" as separate. In glTF and most JSON tools, vectors are stored as arrays.

```suggestion
      "id": "someShapeID124135",
      "position": [10, 10],
      "properties": [{
```

Reply:
- Combining x,y and w,h as arrays is fine. @OrionReed and I had discussed this at the beginning and chose to separate them in ignorance of a precedent.
- `size` doesn't immediately resolve to `width` and `height` in my mind, but if that's what gltf is doing, I'd rather follow in an existing track!